src/Security/Group.php

```diff
@@ -377,6 +377,68 @@ public function collateAncestorIDs()
     }
 
     /**
+     * Check if the group is a child of the given group or any parent groups
+     *
+     * @param string|int|Group $group Group instance, Group Code or ID
+     * @return bool Returns TRUE if the Group is a child of the given group, otherwise FALSE
+     */
+    public function inGroup($group)
+    {
+        $groupCheckObj = $this->identifierToGroup($group);
+
+        if (!$groupCheckObj) {
+            return false;
+        }
+
+        return in_array($groupCheckObj->ID, $this->collateAncestorIDs());
+    }
+
+    /**
+     * Check if the group is a child of the given groups or any parent groups
+     *
+     * @param (string|int|Group)[] $groups
+     * @param bool $requireAll set to TRUE if must be in ALL groups, or FALSE if must be in ANY
+     * @return bool Returns TRUE if the Group is a child of any of the given groups, otherwise FALSE
+     */
+    public function inGroups($groups, $requireAll = false)
+    {
+        $ancestorIDs = $this->collateAncestorIDs();
+        $candidateIDs = [];
+        foreach ($groups as $group) {
+            $candidate = $this->identifierToGroup($group);
+            if (!$candidate) {
+                continue;
+            }
+            $candidateIDs[] = $candidate->ID;
+        }
+        if ($requireAll) {
+            return array_intersect($candidateIDs, $ancestorIDs) === $ancestorIDs;
+        }
+        return count(array_intersect($candidateIDs, $ancestorIDs)) > 0;
+    }
+
+    /**
+     * Turn a string|int|Group into a Group object
+     *
+     * @param string|int|Group $group Group instance, Group Code or ID
+     * @return Group|bool Returns FALSE if the Group doesn't exit, otherwise returns the Group Object
+     */
+    protected function identifierToGroup($group)
+    {
+        if (is_numeric($group)) {
+            $groupCheckObj = Group::get()->byID($group);
+        } elseif (is_string($group)) {
+            $groupCheckObj = Group::get()->filter('Code', $group)->first();
```

Comment:
If you require a group `somecode` that doesn't exist, your code will silently ignore it and pass.